pkg/internal/ebpf/tracer_linux.go

```
@@ -222,21 +222,19 @@ func (pt *ProcessTracer) loadTracers() error {
 	defer loadMux.Unlock()
 
 	var log = ptlog()
-	anyLoaded := false
+
 	for _, p := range pt.Programs {
 		if err := pt.loadTracer(p, log); err != nil {
-			log.Warn("couldn't load tracer", "error", err)
-		} else {
-			anyLoaded = true
+			log.Warn("couldn't load tracer", "error", err, "required", p.Required())
+
+			if p.Required() {
```

@grcevski with regards to your other comment, wouldn't this line prevent it from crashing? `Required()` would be false in the case of `tpinjector`, so the error is just ignored, but I could be missing something.

---

Check the failure logs, you'll see.